src/Artsy/Analytics/TrackingContext.tsx

```diff
@@ -1,22 +1,55 @@
 import { track as _track } from "Artsy/Analytics"
 import * as AnalyticsSchema from "Artsy/Analytics/Schema"
 import React from "react"
-import { TrackingProp } from "react-tracking"
 
 type Trackables = AnalyticsSchema.Trackables
 
 interface TrackingContextProps {
-  tracking?: TrackingProp<Trackables>
+  trackEvent?<P>(trackingProps: Partial<Trackables> & P): void
```

Comment:
@ds300, @alloy - This approach simplifies things, and scopes to our needs (vs going up a level into react-tracking); however, I can't for the life of me figure out how to get this bit to work as I would like it to. In [the previous iteration](https://github.com/artsy/reaction/pull/2405/files#r283638766) it validated correctly and would force the user to expand the scope with a generic if something not a part of `Trackables` was passed. 

Meaning: 

```tsx
trackEvent({
  unknownProp: ...
}) 
```
should throw a type error, because we haven't passed `P`; e.g.,  

```tsx
trackEvent<{ unknownProp: string }>({
  unknownProp: ...
}) 
```

If I remove `P` from the interface above it correctly validates, and will check against `Partial<Trackables>`. 

I would like to to a) make sure the user isn't mistyping something from `Trackables`; while b) enforcing that they pass additional data for one-off props that don't belong in `Trackables`.

Reply:
Hmmm tough one, this is the best I could do:

https://www.typescriptlang.org/play/#src=interface%20Trackables%20%7B%0D%0A%20%20%20%20trackableProp%3A%20string%0D%0A%20%20%20%20otherTrackableProp%3A%20string%0D%0A%7D%0D%0A%0D%0Adeclare%20function%20trackEvent%3CP%20extends%20Trackables%3E(trackingProps%3A%20Partial%3CP%3E)%3A%20void%0D%0A%0D%0A%2F%2F%20doesn't%20expect%20all%20trackable%20props%0D%0AtrackEvent(%7B%20trackableProp%3A%20%22%22%20%7D)%0D%0A%2F%2F%20complains%20if%20you%20provide%20one%20that%20doesn't%20exist%0D%0AtrackEvent(%7B%20trackableProp%3A%20%22%22%2C%20newProp%3A%203%20%7D)%0D%0A%2F%2F%20doesn't%20complain%20if%20you%20explicitly%20add%20more%0D%0AtrackEvent%3CTrackables%20%26%20%7BnewProp%3A%20number%7D%3E(%7B%20trackableProp%3A%20%22%22%2C%20newProp%3A%203%20%7D)

Reply:
This will work -- we could do something like 
```ts
export type TrackingEvent<P> = Trackables & P

... 

trackEvent<TrackingEvent<{newProp: number}>>({ trackableProp: "", newProp: 3 })
```